### ISSUE TITLE
Test failure baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid6\\bindhandleinvalid6.cmd

### ISSUE THREAD

**Issue by VincentBu** [dotnet/runtime#66273]
Run: [runtime-coreclr outerloop 20220306.2](https://dev.azure.com/dnceng/public/_build/results?buildId=1647830&view=ms.vss-test-web.build-test-results-tab&runId=45502822&resultId=100064&paneView=debug)

Failed test:
```
R2R-CG2 windows arm64 Checked no_tiered_compilation @ Windows.10.Arm64v8.Open

- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid6\\bindhandleinvalid6.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid5\\bindhandleinvalid5.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid4\\bindhandleinvalid4.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid3\\bindhandleinvalid3.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid\\bindhandleinvalid.cmd

R2R-CG2 windows arm64 Checked @ Windows.10.Arm64v8.Open

- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid6\\bindhandleinvalid6.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid4\\bindhandleinvalid4.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid5\\bindhandleinvalid5.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid3\\bindhandleinvalid3.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid\\bindhandleinvalid.cmd

R2R-CG2 windows arm Checked @ Windows.10.Arm64v8.Open

- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid6\\bindhandleinvalid6.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid4\\bindhandleinvalid4.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid5\\bindhandleinvalid5.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid3\\bindhandleinvalid3.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid\\bindhandleinvalid.cmd

CoreCLR windows x86 Checked no_tiered_compilation @ Windows.10.Amd64.Open

- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid6\\bindhandleinvalid6.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid4\\bindhandleinvalid4.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid5\\bindhandleinvalid5.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid3\\bindhandleinvalid3.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid\\bindhandleinvalid.cmd


R2R-CG2 windows x64 Checked @ Windows.10.Amd64.Open

- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid6\\bindhandleinvalid6.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid4\\bindhandleinvalid4.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid5\\bindhandleinvalid5.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid3\\bindhandleinvalid3.cmd
- baseservices\\threading\\threadpool\\bindhandle\\bindhandleinvalid\\bindhandleinvalid.cmd
```
**One of error message:**
```
Could Not Find D:\h\w\B06A09A6\w\A88D0A0C\e\baseservices\threading\threadpool\bindhandle\bindhandleinvalid6\IL-CG2\composite-r2r.dll
Could Not Find D:\h\w\B06A09A6\w\A88D0A0C\e\baseservices\threading\threadpool\bindhandle\bindhandleinvalid6\bindhandleinvalid6.dll.rsp

Return code:      1
Raw output file:      D:\h\w\B06A09A6\w\A88D0A0C\uploads\Reports\baseservices.threading\threadpool\bindhandle\bindhandleinvalid6\bindhandleinvalid6.output.txt
Raw output:
BEGIN EXECUTION
bindhandleinvalid6.dll
1 file(s) copied.
Response file: D:\h\w\B06A09A6\w\A88D0A0C\e\baseservices\threading\threadpool\bindhandle\bindhandleinvalid6\bindhandleinvalid6.dll.rsp
D:\h\w\B06A09A6\w\A88D0A0C\e\baseservices\threading\threadpool\bindhandle\bindhandleinvalid6\IL-CG2\bindhandleinvalid6.dll
-o:D:\h\w\B06A09A6\w\A88D0A0C\e\baseservices\threading\threadpool\bindhandle\bindhandleinvalid6\bindhandleinvalid6.dll
--targetarch:arm64
--verify-type-and-field-layout
--method-layout:random
-r:D:\h\w\B06A09A6\p\System..dll
-r:D:\h\w\B06A09A6\p\Microsoft..dll
-r:D:\h\w\B06A09A6\p\mscorlib.dll
-r:D:\h\w\B06A09A6\p\netstandard.dll
-O
" "dotnet" "D:\h\w\B06A09A6\p\crossgen2\crossgen2.dll" @"D:\h\w\B06A09A6\w\A88D0A0C\e\baseservices\threading\threadpool\bindhandle\bindhandleinvalid6\bindhandleinvalid6.dll.rsp"   -r:D:\h\w\B06A09A6\w\A88D0A0C\e\baseservices\threading\threadpool\bindhandle\bindhandleinvalid6\IL-CG2*.dll"
Emitting R2R PE file: D:\h\w\B06A09A6\w\A88D0A0C\e\baseservices\threading\threadpool\bindhandle\bindhandleinvalid6\bindhandleinvalid6.dll
"D:\h\w\B06A09A6\p\corerun.exe" -p "System.Reflection.Metadata.MetadataUpdater.IsSupported=false"  bindhandleinvalid6.dll
BindHandle call succeeded
Got wrong error: System.ApplicationException: Error in the application.
at System.Threading.PortableThreadPool.RegisterForIOCompletionNotifications(IntPtr handle)
at System.Threading.ThreadPool.BindHandle(SafeHandle osHandle)
at BindHandle1.RunTest()
Expected: 100
Actual: 99
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:

set CORE_ROOT=D:\h\w\B06A09A6\p
D:\h\w\B06A09A6\w\A88D0A0C\e\baseservices\threading\threadpool\bindhandle\bindhandleinvalid6\bindhandleinvalid6.cmd
Expected: True
Actual:   False


Stack trace
   at baseservices_threading._threadpool_bindhandle_bindhandleinvalid6_bindhandleinvalid6_._threadpool_bindhandle_bindhandleinvalid6_bindhandleinvalid6_cmd()
```

**Comment by mangod9** [comment]
@kouvel, since this looks related to the recent IOCompletion changes -- looks to be reproing consistently.

**Comment by BruceForstall** [comment]
When fixed, re-enable the tests disabled with #66303
